app/src/main/java/com/example/magiclines/ui/selectionLevel/SelectLevelFragment.kt

```diff
@@ -64,16 +69,36 @@ class SelectLevelFragment : BaseFragment<FragmentSelectionLevelBinding, SelectLe
 
 //        initViewModel()
 
-        observeData()
-        scheduleEnergyWork()
+
+        if (isLoading){
+            Log.e("TAG", "onViewCreated: load", )
+            viewModel.getDataFiltered(levelAdapter!!, currentCategory, "")
+            isLoading = false
+        }
+
+        viewModel.filteredLevels.observe(viewLifecycleOwner) { filteredLevels ->
+            levelAdapter!!.submitList(filteredLevels)
+        }
+
+        viewModel.currentCategory.observe(viewLifecycleOwner) { categoryIndex ->
+            categoryAdapter!!.setCurrentCategory(categoryIndex)
```

Comment:
Fix tương tự. Dùng ? thay vì !!
```suggestion
            categoryAdapter?.setCurrentCategory(categoryIndex)
```